core/domain/role_domain_test.py

```diff
@@ -0,0 +1,134 @@
+        to all the keys
+        """
+        for i in self.HIERARCHY:
+            self.assertTrue(isinstance(self.HIERARCHY[i], list))
+
+        for i in self.ACTIONS:
+            self.assertTrue(isinstance(self.ACTIONS[i], list))
+
+    def test_valid_names(self):
+        """strings with UPPERCASE alphabets and underscores are allowed. This
+        test checks whether all entries comply with this rule.
+        """
+
+        def valid_character(character):
+            if (character in string.ascii_uppercase) or (character == '_'):
+                return True
+            return False
+
+        for role_name in self.HIERARCHY:
+            for i in role_name:
+                self.assertTrue(valid_character(i))
+
+            for action_name in self.HIERARCHY[role_name]:
+                for i in action_name:
+                    self.assertTrue(valid_character(i))
+
+    def test_every_dict_entry_is_string(self):
+        """test that all keys and values(elements in lists) in ROLE_HIERARCHY
+        and ROLE_ACTIONS are string.
+        """
+
+        # checking the keys in ROLE_HIERARCHY.
+        for i in self.HIERARCHY:
+            self.assertTrue(isinstance(i, str))
+
+            # checking the values in list corresponding to key.
+            for j in self.HIERARCHY[i]:
+                self.assertTrue(isinstance(j, str))
+
+        # checking the keys in ACTIONS.
+        for i in self.ACTIONS:
+            self.assertTrue(isinstance(i, str))
+
+            # checking the values in list corresponding to key.
+            for j in self.ACTIONS[i]:
+                self.assertTrue(isinstance(j, str))
+
+    def test_valid_parents(self):
+        """test that all the roles present in value list for any key in
+        ROLE_HIERARCHY are valid(i.e there exists a key with that name)
+        """
+        valid_roles = self.HIERARCHY.keys()
+
+        for i in self.HIERARCHY:
+            for j in self.HIERARCHY[i]:
+                self.assertTrue(j in valid_roles)
+
+    def test_no_cycles(self):
+        """Visits each role and checks that there is no cycle from that
+        role"""
+        visited = set()
+
+        def check_cycle(source, parents):
+            """checks if there is a cycle starting from source
+            args : source -> the starting role to check
+                   parents -> list of neighbours
+            """
+            if len(parents) == 0:
+                return
+
+            for i in parents:
+                self.assertNotEqual(i, source)
+                if i not in visited:
+                    visited.add(i)
+                    check_cycle(source, self.HIERARCHY[i])
+
+        for i in self.HIERARCHY:
+            visited = set()
+            check_cycle(i, self.HIERARCHY[i])
+
+    def test_get_all_actions_is_working(self):
+        """test that get_all_actions is working correctly."""
+
+        # case when wrong input is given
+        with self.assertRaisesRegexp(Exception,
```

Comment:
```
with self.assertRaisesRegexp(
    Exception, 'no role with ...'):
```
This indentation shows error in my subline plugin linter, so I did that.

But, it doesn't show error with pre_commit_linter.py so I changed it now. 

done.